pkg/cache/server/config.go

```diff
@@ -136,8 +136,11 @@ func NewConfig(opts *cacheserveroptions.CompletedOptions, optionalLocalShardRest
 	}
 
 	serverConfig.Config.BuildHandlerChainFunc = func(apiHandler http.Handler, genericConfig *genericapiserver.Config) (secure http.Handler) {
-		apiHandler = genericapiserver.DefaultBuildHandlerChainFromAuthz(apiHandler, genericConfig)
-		apiHandler = genericapiserver.DefaultBuildHandlerChainBeforeAuthz(apiHandler, genericConfig)
+
+		apiHandler = genericapiserver.DefaultBuildHandlerChainFromAuthzToCompletion(apiHandler, genericConfig)
```

Comment:
Not sure we want to catch impersonation errors in cache server too?

Reply:
How do we authorize the cache-server? Only `system:masters` in the first place?

Reply:
right. I think we can revisit this post merge. WIll create issue to check this.